packager/hls/base/master_playlist.cc

```diff
@@ -465,38 +477,35 @@ void AppendPlaylists(const std::string& default_audio_language,
     }
   }
 
-  // convert the std::map to std::list and reorder it if indexes were provided
-  std::list<std::pair<std::string, std::list<const MediaPlaylist*>>>
-      audio_groups_list(audio_playlist_groups.begin(),
-                        audio_playlist_groups.end());
-  std::list<std::pair<std::string, std::list<const MediaPlaylist*>>>
-      subtitle_groups_list(subtitle_playlist_groups.begin(),
-                           subtitle_playlist_groups.end());
   if (has_index) {
-    audio_groups_list.sort(GroupOrderFn);
-    for (const auto& group : audio_groups_list) {
-      std::list<const MediaPlaylist*> group_playlists = group.second;
-      group_playlists.sort(ListOrderFn);
-    }
-    subtitle_groups_list.sort(GroupOrderFn);
-    for (const auto& group : subtitle_groups_list) {
-      std::list<const MediaPlaylist*> group_playlists = group.second;
-      group_playlists.sort(ListOrderFn);
-    }
-    video_playlists.sort(ListOrderFn);
-    iframe_playlists.sort(ListOrderFn);
+    video_playlists.sort(PlaylistOrderFn);
+    iframe_playlists.sort(PlaylistOrderFn);
   }
 
   if (!audio_playlist_groups.empty()) {
     content->append("\n");
-    BuildMediaTags(audio_groups_list, default_audio_language, base_url,
-                   content);
+    std::list<MediaTagslist> audio_playlists =
```

Comment:
above we group playlists into group using `audio_playlist_groups[GetGroupId(*playlist)].push_back(playlist);` so that the groups can be sorted, but now we're no longer doing that, instead flattening the group and sorting that. We should probably insert the right structs into a list in that first `for (const MediaPlaylist* playlist : playlists) {` loop then we can sort each list, and no longer keep things grouped.

Reply:
Are you suggesting removing `std::map<std::string, std::list<const MediaPlaylist*>> audio_playlist_groups` and replacing it with a new `std::list<MediaTagslist> audio_playlists`?  The `audio_playlist_groups` is still used by BuildVariants(), so that and BuildMediaTags() would need to be refactored.

I initially started down that route but felt that the code in BuildMediaTags() was easier to follow when grouped first.  I could re-address this if you think otherwise.

Reply:
@SteveR-PMP ah yes, `BuildVariants()` still operates on groups. It still feels that using `std::list<MediaTagslist> audio_playlists`, refactoring `BuildMediaTags()` to use this (and call `BuildMediaTag()` within as before) and deferring any necessary grouping to `BuildVariants()` would be cleaner.

Reply:
do we want to do this refactor now @cosmin  ? me or @SteveR-PMP  haven't really given this a try yet and test again the combinations.

Reply:
@vish91 I do think this needs a refactoring, one way or another.